### PR TITLE
Add RidSpecific, Make Apps RID Specific By Default & RID Now FDD By Default in 8.0+

### PR DESCRIPTION
**Pull request by nagilson**
For https://github.com/dotnet/sdk/issues/26031 
7.0 backport https://github.com/dotnet/sdk/pull/28376

# Adds `RuntimeSpecific`:
1. Makes TFM >= 8.0 Exe apps Infer RID automatically (FDD) 
2. RIDS no longer imply Self-Contained for TFM >= 8.0
3. Allows opt out of the inferred RID behavior (FDDefault) by setting `RuntimeSpecific` to false, but no opt-out to not implying SC with RIDs in TFM 8.0
4. Warns for apps that didn't set SC but set RIDs in TFM < 8 that they need to set SC explicitly to get the same behavior with TFM 8.0. Warning will go away after SC is set, either to true or false.

# Based on .NET 8.0 Breaking Changes Notes:
* Make RID Specific Apps FDD (NOT SC) by default instead of SC
* Make EXE Apps Rid Specific By Default

# Tested Behavior:
* RidNoLongerImpliesSelfContainedForTFMOverSeven
* RidImpliesSelfContainedButWarnsIfNotSetForTFMUnderEight
* RidSpecificMakesExecutableAppsAboveTFMSevenRidSpecificByDefault
* RidSpecificAllowsOptOutRIDlessAppsOnExecutableAppsForTFMAboveSeven
* SelfContainedPlusRuntimeIdentifierDoesNotTriggerEightTFMFDDBreakingChangeWarning
* RuntimeIdentifiersDisablesRuntimeSpecificFDDBehavior

# Questions:
- Why does `TargetFrameworkInference` sometimes not run? I accounted for that, but don't understand the contexts.
- Do we care to set a rid with `RidSpecific` by default if `RuntimeIdentifiers` is set? ATM, I made it so `RuntimeSpecific` doesn't do anything if `RuntimeIdentifiers` is set, nor will apps with RuntimeIdentifiers be FDD. 
- Could we use a more specific RID?

Verify: 
 Make sure you can still specify -r runtime without it being overriden.
 Make sure you can still specify self-contained to true even if you do /p:SelfContained=true.

### CODE REVIEW

**Comment by nagilson** [comment]
Still blocked by https://github.com/dotnet/sdk/pull/28563

**Comment by nagilson** [comment]
This is no longer blocked.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets [hunk 1]
```diff
@@ -61,6 +61,14 @@ Copyright (c) .NET Foundation. All rights reserved.
     <RuntimeIdentifier Condition="'$(PlatformTarget)' == 'x86' or '$(PlatformTarget)' == ''">win7-x86</RuntimeIdentifier>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(RuntimeSpecific)' == '' and '$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '8.0' and '$(OutputType)' == 'Exe'">
```
Comment (teo-tsirpanis) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:64]:
```suggestion
  <PropertyGroup Condition="'$(RuntimeSpecific)' == '' and '$(_TargetFrameworkVersionWithoutV)' != '' and '$(_TargetFrameworkVersionWithoutV)' >= '8.0' and ('$(OutputType)' == 'Exe' or '$(OutputType)' == 'WinExe')">
```

**Comment by nagilson** [comment]
We aren't doing this proposal anymore, there is a new proposal

**Comment by teo-tsirpanis** [comment]
Where is that new proposal?

**Pull request closed without merging**